Bolts/Common/BFTask.m

```diff
@@ -33,24 +33,30 @@ @interface BFTask () {
 @property (atomic, assign, readwrite, getter=isCompleted) BOOL completed;
 
 @property (nonatomic, strong) NSObject *lock;
-@property (nonatomic, strong) NSCondition *condition;
+@property (nonatomic, assign) dispatch_semaphore_t completedSemaphore;
```

Comment:
Yikes. I definitely should have noticed that when I added the conditional compilation. Thanks for noticing that.